app/liquid_tags/link_tag.rb

```diff
@@ -26,7 +26,10 @@ def article_hash(slug)
     path = Addressable::URI.parse(slug).path
     path.slice!(0) if path.starts_with?("/") # remove leading slash if present
     path.slice!(-1) if path.ends_with?("/") # remove trailing slash if present
-    Addressable::Template.new("{username}/{slug}").extract(path)&.symbolize_keys
+    extracted_hash = Addressable::Template.new("{username}/{slug}").extract(path)&.symbolize_keys
+    raise StandardError, "This URL is not an article link" unless extracted_hash
```

Comment:
This is definitely a step up from our previous iteration! I would suggest also displaying the URL itself, like so:
```
    raise StandardError, "This URL is not an article link: {% link #{slug} %}" unless extracted_hash
```

I _think_ `slug` is the correct variable, in order to render the actual input of the user.